Sources/Sentry/SentryDependencyContainer.m

```diff
@@ -0,0 +1,60 @@
+#import <Foundation/Foundation.h>
+#import <SentryAppStateManager.h>
+#import <SentryClient+Private.h>
+#import <SentryCrashAdapter.h>
+#import <SentryDefaultCurrentDateProvider.h>
+#import <SentryDependencyContainer.h>
+#import <SentryHub.h>
+#import <SentrySDK+Private.h>
+#import <SentrySysctl.h>
+#import <SentryThreadWrapper.h>
+
+@implementation SentryDependencyContainer
+
+static SentryDependencyContainer *instance;
+
++ (instancetype)sharedInstance
+{
+    if (instance == nil) {
+        instance = [[self alloc] init];
+    }
```

Comment:
l: This could cause a race condition?

Reply:
Good point, here we could do a double-check lock just to make sure we lock only for the first initialization and avoid any contention after that